File: maximum-subarray/dynamic-programming-solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        l = len(nums)
        temp_sum = 0
        iter2 = 0
        max1 = nums[0]
        
        if(l == 1):
            return nums[0]
        
        for i in range(1,l):
            if(nums[i] < nums[i]+nums[i-1]):
                nums[i] = nums[i-1] + nums[i]
     
            if(nums[i] > max1):
                max1 = nums[i] 
                
                    
        return max1

        

File: maximum-subarray/dynamic-programming-solution.cpp
//https://leetcode.com/problems/maximum-subarray/
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max=nums[0];
        int l = nums.size();
        
        if(l == 1)
        {
            
            return *nums.begin();
        }
        
        for(int i = 1;i<l;i++)
        {
            
            if(nums[i] < nums[i] + nums[i-1])
            {
                
                nums[i] = nums[i] + nums[i-1];

                
            }
            
            if(nums[i] > max)
            {
                        
                max = nums[i];
                        
            }
            
            
        }


        return max;
        
    }
    
};
